src/Markdig/Helpers/HtmlHelper.cs

```diff
@@ -465,6 +465,24 @@ private static bool TryParseHtmlTagProcessingInstruction(ref StringSlice text, r
         }
     }
 
+    /// <summary>
+    /// Destructively unescape a string: remove backslashes before punctuation or symbol characters.
+    /// </summary>
+    /// <param name="text">The string data that will be changed by unescaping any punctuation or symbol characters.</param>
+    /// <param name="removeBackSlash">if set to <c>true</c> [remove back slash].</param>
+    /// <returns>Unescaped text, or null if <paramref name="text"/> is null.<returns>
+    public static string? UnescapeNullable(string? text, bool removeBackSlash = true)
```

Comment:
I don't think this helper is valuable, can we do a simple null check at the one call site instead?